src/compiler/scala/tools/nsc/backend/jvm/BCodeSkelBuilder.scala

```diff
@@ -263,7 +261,7 @@ abstract class BCodeSkelBuilder extends BCodeHelpers {
     }
 
     def addClassFields(): Unit = {
-      for (f <- fieldSymbols(claszSymbol)) {
+      fieldSymbolsForeach(claszSymbol) { f =>
```

Comment:
I don't think you can safely use the reversed-order iterator here as the the fields will be added to the `ClassNode` in the wrong order.